component/src/test/java/org/wso2/extension/siddhi/gpl/execution/geo/function/GeoWithinDistanceTestCase.java

```diff
@@ -61,15 +73,37 @@ public void testGeometry() throws Exception {
             public void receive(long timeStamp, Event[] inEvents, Event[] removeEvents) {
                 for (Event event : inEvents) {
                     logger.info(event);
-                    Boolean isWithin = (Boolean) event.getData(0);
-                    AssertJUnit.assertEquals(expectedResult.get(eventCount++), isWithin);
+                    count.incrementAndGet();
+                    if (count.get() == 1) {
+                        Boolean isWithin = (Boolean) event.getData(0);
+                        AssertJUnit.assertEquals(expectedResult.get(eventCount++), isWithin);
+                        eventArrived = true;
+                    }
+                    if (count.get() == 2) {
+                        Boolean isWithin = (Boolean) event.getData(0);
+                        AssertJUnit.assertEquals(expectedResult.get(eventCount++), isWithin);
+                        eventArrived = true;
+                    }
+                    if (count.get() == 3) {
+                        Boolean isWithin = (Boolean) event.getData(0);
+                        AssertJUnit.assertEquals(expectedResult.get(eventCount++), isWithin);
+                        eventArrived = true;
+
+                    }
+                    if (count.get() == 4) {
+                        Boolean isWithin = (Boolean) event.getData(0);
+                        AssertJUnit.assertEquals(expectedResult.get(eventCount++), isWithin);
+                        eventArrived = true;
+
+                    }
                 }
             }
         });
         siddhiAppRuntime.start();
         generateEvents(siddhiAppRuntime);
-        Thread.sleep(1000);
-        AssertJUnit.assertEquals(expectedResult.size(), eventCount);
+        SiddhiTestHelper.waitForEvents(100, 4, count, 60000);
```

Comment:
I think we need to stop the siddhiAppRuntime. WDYT?

Reply:
Fixed by 26e6366